bldsys/cmake/global_options.cmake

```diff
@@ -33,15 +33,10 @@ endif()
 
 if(APPLE)
     cmake_minimum_required(VERSION 3.24)
+    set(VKB_ENABLE_PORTABILITY ON CACHE BOOL "Enable portability enumeration and subset features in the framework.  This is required to be set when running on Apple platforms." FORCE)
```

Comment:
I think the original method might be preferred.  If someone is using vulkan less than 1.3 then they are on a build that doesn't yet have the portability extension.  Which would mean their app would go from working to not working with this mandated.

Reply:
My change here was motivated by feedback from @SaschaWillems and @asuessenbach.  They both felt that using `VK_ENABLE_BETA_EXTENSIONS` as a guard in the code base was confusing and overloaded.  I tend to agree.  This define should be used as input for configuring Vulkan and not changing code in Vulkan-Samples, specifically around use of the `VK_KHR_portability_subset` device extension and features.  The project should be able to support future uses of Vulkan beta extensions other than for just portability.  I wanted to make a simple change to fix this without introducing a new define into the VKB_ configuration parameters.

What I ended up doing was using the existing `VKB_ENABLE_PORTABILITY` define to switch on both the `VK_KHR_portability_enumeration` instance extension and the `VK_KHR_portability_subset` device extension/features on Apple platforms (can also be used on any other portability platform).  However, I did this with a difference compared to the existing code base: Rather than relying on static config within cmake, I now test for existence at runtime in the code.  When running against an older SDK without the enumeration extension, it will not be found and therefore not enabled.  I have also tested this against MoltenVK standalone which does not implement the enumeration extension since that is a loader-only capability.  Note that build-time concerns are not a problem since Vulkan-Samples ships with Vulkan Headers 277, where the enumeration extension is defined.

Lastly, the old cmake test for Vulkan SDK 1.3.x was not strictly correct.  The `VK_KHR_portability_enumeration` extension was introduced in SDK 1.3.211, and made mandatory in 1.3.216.  Any 1.3 SDKs earlier than 211 will fail under the current implementation (i.e. extension not present).  I am not sure if this is much of a practical concern nowadays, but it's a detail worth noting for comparison purposes.

Reply:
Perfect!  Thanks for the work and getting the right solution!  I'm sold on this.